src/routers/user.ts

```diff
@@ -94,22 +117,50 @@ user.get(
   '/users',
   userQuerySubstringLimitValidator,
   methodLoggerMiddleware,
-  async (req: ValidatedRequest<QuerySubstringLimitSchema>, res: Response, next: NextFunction) => {
+  checkTokenMiddleware,
+  async (
+    req: ValidatedRequest<QuerySubstringLimitSchema>,
+    res: Response,
+    next: NextFunction
+  ) => {
     try {
       const { loginsubstring } = req.query;
       const { limit } = req.query;
 
       const result = await userService.getUsers(loginsubstring, limit);
       res.status(200).json({ users: result });
     } catch (e) {
-      next(e)
+      next(e);
     }
   }
 );
 
-user.get('/error', methodLoggerMiddleware, function() {
-  //throw new Error("I AM UNHANDLED EXEPTION>>>>>");
-  //new Promise((res,rej)=>{rej('I AM UNHANDLED PROMISE REJECTION>>>>>')});
-});
+user.post(
+  '/user/authenticate',
+  userBodyCredentialsValidator,
+  async (
+    req: ValidatedRequest<GetUserByCredentialsBodySchema>,
+    res: Response,
+    next: NextFunction
+  ) => {
+
+    try {
+      const { login, password } = req.body;
+
+      const result = await userService.getUserByCredentials(login, password);
+
+      if (result instanceof User) {
+        const payload = { sub: 'api access', userId: result.id };
+        const secret = process.env.SECRET as string;
+        const token = jwt.sign(payload, secret, { expiresIn: '30000ms' });
```

Comment:
Better not to use magical numbers. Create a constant for 30000 and use it here.

Reply:
fixed